### PR TITLE
Check for gregset_t before using it.

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
@kiranchandramohan, this is ready for review

**Comment by kaadam** [comment]
HI @isuruf, thanks for your contribution. The change looks good to me (informal).

**Review by gklimowicz — APPROVED** [review]
These look good to me. They didn't hurt my OpenPOWER LLVM 9 an 10 builds that I could see.

**Review by shivaramaarao — COMMENTED** [review]
can you please educate me on  gregset_t?. what is this used for?

thanks.

**Review by bryanpkc — COMMENTED** [review]
Can you define a macro to replace `gregset_t`, instead of creating alternative definitions of `dumpregs` and `getRegs`? I think that would lead to a smaller diff.

**Comment by kaadam** [comment]
@isuruf Could you squash these commits into one? Thanks. @bryanpkc @shivaramaarao Could you look at it again?

File: runtime/flangrti/aarch64-Linux/dumpregs.c
Comment (bryanpkc) [inline runtime/flangrti/aarch64-Linux/dumpregs.c:128]: Is it possible to replace these empty functions with void macros? e.g.
```c
#if defined(HAVE_GREGSET_T)
...
#else
#define dumpregs ((void *) 0)
#define getRegs ((void *) 0)
#endif
```

Comment (isuruf) [reply runtime/flangrti/aarch64-Linux/dumpregs.c:128]: I could but then, I'll have to change other places where `dumpregs` is called with ifdefs.

Comment (isuruf) [reply runtime/flangrti/aarch64-Linux/dumpregs.c:128]: since `void *` is not a function.

Comment (isuruf) [reply runtime/flangrti/aarch64-Linux/dumpregs.c:128]: ping on this

Comment (bryanpkc) [reply runtime/flangrti/aarch64-Linux/dumpregs.c:128]: `dumpregs` and `getRegs` are only called in one file, so it's not a big deal either way. OK.

**Comment by isuruf** [comment]
@bryanpkc, can you check again?

**Comment by kaadam** [comment]
@bryanpkc Could you look at it again? :)

**Review by bryanpkc — APPROVED** [review]
Casting a `mcontext_t *` to `gregset_t *` is ugly, but I understand that's not your fault.

**Comment by bryanpkc** [comment]
Could you squash the commits, and wait for @shivaramaarao 's approval?

**Comment by isuruf** [comment]
@shivaramaarao, can you review?

**Comment by isuruf** [comment]
ping

**Comment by kiranchandramohan** [comment]
Can @kaadam or @xoviat approve this if OK?

**Review by kaadam — APPROVED** [review]
LGTM

**Pull request merged**
